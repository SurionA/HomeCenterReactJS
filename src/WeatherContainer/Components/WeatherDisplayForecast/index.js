import React from 'react';
import { Grid, Col } from 'react-bootstrap';
import PropTypes from 'prop-types';

import ForecastBlock from './Components/ForecastBlock';

function WeatherDisplayForecast({ foreCastWeatherData }) {
  return (
    <Grid>
      {foreCastWeatherData.map(weatherData => (
        <Col xs={12} md={4} key={weatherData.dt}>
          <ForecastBlock weatherData={weatherData} />
        </Col>
      ))}
    </Grid>
  );
}

WeatherDisplayForecast.propTypes = {
  foreCastWeatherData: PropTypes.arrayOf(
    PropTypes.shape({
      temperature_max: PropTypes.number,
      temperature_min: PropTypes.number,
      icon: PropTypes.number,
      dt: PropTypes.number,
      dayShortName: PropTypes.string,
    })
  ),
};

WeatherDisplayForecast.defaultProps = {
  foreCastWeatherData: PropTypes.arrayOf(PropTypes.shape({})),
};

export default WeatherDisplayForecast;
