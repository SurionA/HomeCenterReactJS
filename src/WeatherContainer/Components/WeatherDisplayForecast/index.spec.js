import React from 'react';
import renderer from 'react-test-renderer';
import WeatherDisplayForecast from './';

it('renders correctly WeatherDisplayForecast component', () => {
  const foreCastWeatherData = [
    {
      temperature_max: 16,
      temperature_min: 12,
      icon: 800,
      dt: 123,
      dayShortName: 'Mon',
    },
    {
      temperature_max: 13,
      temperature_min: 9,
      icon: 800,
      dt: 456,
      dayShortName: 'Thus',
    },
    {
      temperature_max: 14,
      temperature_min: 13,
      icon: 800,
      dt: 789,
      dayShortName: 'Wen',
    },
  ];
  const tree = renderer.create(<WeatherDisplayForecast foreCastWeatherData={foreCastWeatherData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
