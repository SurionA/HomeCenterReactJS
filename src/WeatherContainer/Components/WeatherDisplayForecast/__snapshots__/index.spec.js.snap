// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly WeatherDisplayForecast component 1`] = `
<div
  className="container"
>
  <div
    className="col-md-4 col-xs-12"
  >
    <div
      className="row"
    >
      <h3
        className="title"
      >
        Mon
      </h3>
    </div>
    <div
      className="row"
    >
      <div
        className="wi wi-icon-800"
        style={
          Object {
            "fontSize": "4rem",
            "margin": "15px 0",
          }
        }
      />
    </div>
    <div
      className="row"
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      16
      °
    </div>
    <div
      className="row"
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      12
      °
    </div>
  </div>
  <div
    className="col-md-4 col-xs-12"
  >
    <div
      className="row"
    >
      <h3
        className="title"
      >
        Thus
      </h3>
    </div>
    <div
      className="row"
    >
      <div
        className="wi wi-icon-800"
        style={
          Object {
            "fontSize": "4rem",
            "margin": "15px 0",
          }
        }
      />
    </div>
    <div
      className="row"
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      13
      °
    </div>
    <div
      className="row"
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      9
      °
    </div>
  </div>
  <div
    className="col-md-4 col-xs-12"
  >
    <div
      className="row"
    >
      <h3
        className="title"
      >
        Wen
      </h3>
    </div>
    <div
      className="row"
    >
      <div
        className="wi wi-icon-800"
        style={
          Object {
            "fontSize": "4rem",
            "margin": "15px 0",
          }
        }
      />
    </div>
    <div
      className="row"
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      14
      °
    </div>
    <div
      className="row"
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      13
      °
    </div>
  </div>
</div>
`;
